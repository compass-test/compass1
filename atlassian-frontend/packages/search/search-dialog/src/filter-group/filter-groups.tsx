import React from 'react';
import {
  FilterGroupTitle,
  SkeletonContainer,
  FilterGroupContainer,
  FilterResultRowContainer,
} from './filter-group.styled';
import { SkeletonItem, AvatarShape } from '../search-skeleton';

export interface Props {
  /**
   * The title of the filter props
   */
  title: string;
  /**
   * Whether the filters are currently being loaded. This will render a set of placeholder loading filter items.
   */
  isLoading?: boolean;
  /**
   * The shape of the filter icons / avatars. This is used with the placeholder loading filter items and preferably match the icon shape of the actual filter items.
   * This defaults to no avatar.
   */
  avatarShape?: AvatarShape;
}

export const ColumnFilterGroup: React.FunctionComponent<Props> = ({
  children,
  title,
  isLoading,
  avatarShape,
}) => {
  return (
    <FilterGroupContainer>
      <FilterGroupTitle>{title}</FilterGroupTitle>
      {!isLoading ? (
        children
      ) : (
        <SkeletonContainer>
          <SkeletonItem avatarShape={avatarShape} isShimmering />
          <SkeletonItem avatarShape={avatarShape} isShimmering />
          <SkeletonItem avatarShape={avatarShape} isShimmering />
        </SkeletonContainer>
      )}
    </FilterGroupContainer>
  );
};

export const RowFilterGroup: React.FunctionComponent<Props> = ({
  children,
  title,
  isLoading,
  avatarShape = 'none',
}) => {
  return (
    <FilterGroupContainer>
      <FilterGroupTitle>{title}</FilterGroupTitle>
      <FilterResultRowContainer>
        {!isLoading ? (
          children
        ) : (
          <SkeletonContainer>
            <SkeletonItem avatarShape={avatarShape} isShimmering />
          </SkeletonContainer>
        )}
      </FilterResultRowContainer>
    </FilterGroupContainer>
  );
};
