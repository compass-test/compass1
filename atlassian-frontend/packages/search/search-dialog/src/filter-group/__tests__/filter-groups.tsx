import React from 'react';
import { shallow } from 'enzyme';
import { ColumnFilterGroup, RowFilterGroup } from '../filter-groups';

describe('<ColumnFilterGroup />', () => {
  it('matches snapshot when not loading', () => {
    const wrapper = shallow(
      <ColumnFilterGroup title="title text" isLoading={false}>
        <input type="checkbox" />
      </ColumnFilterGroup>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot when not loading with multiple children', () => {
    const wrapper = shallow(
      <ColumnFilterGroup title="title text" isLoading={false}>
        <input type="checkbox" />
        <input type="checkbox" />
      </ColumnFilterGroup>,
    );

    expect(wrapper.find('input')).toHaveLength(2);
    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot when loading', () => {
    const wrapper = shallow(
      <ColumnFilterGroup title="title text" isLoading>
        <input type="checkbox" />
      </ColumnFilterGroup>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('is empty when no children are present', () => {
    const wrapper = shallow(
      <ColumnFilterGroup title="title text" isLoading={false} />,
    );

    expect(wrapper.children().length).toBe(1); // there is one title child component
  });
});

describe('<RowFilterGroup />', () => {
  it('matches snapshot when not loading', () => {
    const wrapper = shallow(
      <RowFilterGroup title="title text" isLoading={false}>
        <input type="checkbox" />
      </RowFilterGroup>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot when not loading with multiple children', () => {
    const wrapper = shallow(
      <RowFilterGroup title="title text" isLoading={false}>
        <input type="checkbox" />
        <input type="checkbox" />
      </RowFilterGroup>,
    );

    expect(wrapper.find('input')).toHaveLength(2);
    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot when loading', () => {
    const wrapper = shallow(
      <RowFilterGroup title="title text" isLoading>
        <input type="checkbox" />
      </RowFilterGroup>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('is empty when no children are present', () => {
    const wrapper = shallow(
      <RowFilterGroup title="title text" isLoading={false} />,
    );

    expect(wrapper.children().length).toBe(2); // one title component and one row container
  });
});
