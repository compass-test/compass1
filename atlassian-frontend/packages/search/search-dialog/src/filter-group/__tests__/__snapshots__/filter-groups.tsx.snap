// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColumnFilterGroup /> matches snapshot when loading 1`] = `
<Styled(div)>
  <Styled(div)>
    title text
  </Styled(div)>
  <Styled(div)>
    <SkeletonItem
      isShimmering={true}
    />
    <SkeletonItem
      isShimmering={true}
    />
    <SkeletonItem
      isShimmering={true}
    />
  </Styled(div)>
</Styled(div)>
`;

exports[`<ColumnFilterGroup /> matches snapshot when not loading 1`] = `
<Styled(div)>
  <Styled(div)>
    title text
  </Styled(div)>
  <input
    type="checkbox"
  />
</Styled(div)>
`;

exports[`<ColumnFilterGroup /> matches snapshot when not loading with multiple children 1`] = `
<Styled(div)>
  <Styled(div)>
    title text
  </Styled(div)>
  <input
    type="checkbox"
  />
  <input
    type="checkbox"
  />
</Styled(div)>
`;

exports[`<RowFilterGroup /> matches snapshot when loading 1`] = `
<Styled(div)>
  <Styled(div)>
    title text
  </Styled(div)>
  <Styled(div)>
    <Styled(div)>
      <SkeletonItem
        avatarShape="none"
        isShimmering={true}
      />
    </Styled(div)>
  </Styled(div)>
</Styled(div)>
`;

exports[`<RowFilterGroup /> matches snapshot when not loading 1`] = `
<Styled(div)>
  <Styled(div)>
    title text
  </Styled(div)>
  <Styled(div)>
    <input
      type="checkbox"
    />
  </Styled(div)>
</Styled(div)>
`;

exports[`<RowFilterGroup /> matches snapshot when not loading with multiple children 1`] = `
<Styled(div)>
  <Styled(div)>
    title text
  </Styled(div)>
  <Styled(div)>
    <input
      type="checkbox"
    />
    <input
      type="checkbox"
    />
  </Styled(div)>
</Styled(div)>
`;
